R/quo.R

```diff
@@ -411,10 +412,23 @@ as_quosure <- function(x, env = caller_env()) {
     new_quosure(x, env)
   }
 }
+
 #' @export
 print.quosure <- function(x, ...) {
-  x <- set_attrs(x, class = "formula")
-  NextMethod()
+  env_type <- env_type(get_env(x))
+
+  if (env_type %in% c("frame", "local")) {
+    addr <- sxp_address(get_env(x))
```

Comment:
I'm not sure either. It does makes the printout look more technical. And the address can still be printed with `get_env()` when needed.